module/zfs/zfs_ioctl.c

```diff
@@ -5815,10 +5815,12 @@ zfs_ioc_clear(zfs_cmd_t *zc)
 
 	/*
 	 * If multihost is enabled, resuming I/O is unsafe as another
-	 * host may have imported the pool.
+	 * host may have imported the pool. Check for remote activity.
 	 */
-	if (spa_multihost(spa) && spa_suspended(spa))
+	if (spa_multihost(spa) && spa_suspended(spa) &&
+	    spa_mmp_remote_host_activity(spa)) {
```

Comment:
The core fix in this PR looks good to me.  However, when manually testing this I wasn't able to resume a MMP suspended pool when I should have been able too.  e.g. when it was never touched by the other host.  Could you retest this locally?  When I get a chance I'll attempt to reproduce it myself and grab some logs to see why.